execute/plugin.go

```diff
@@ -325,9 +341,20 @@ func buildSingleChainReport(
 		return cciptypes.ExecutePluginReportSingleChain{}, 0,
 			fmt.Errorf("unable to construct merkle tree from messages for report (%s): %w", report.MerkleRoot.String(), err)
 	}
-	numMsgs := len(report.Messages)
+
+	// Verify merkle root.
+	hash := tree.Root()
+	if !bytes.Equal(hash[:], report.MerkleRoot[:]) {
+		// For testing purposees, we allow a sentinel root to be used.
+		if report.MerkleRoot.String() != SentinelRoot {
```

Comment:
I am not a fan of having testing code in the prod logic.
What I understand is that you want to do:
```
if computed root does not match report root:
   if report root is a test root:
      we are withing a test so all good
   else:
       error
```


If there is no other way to test this you could instead do:
```
if !p.testMode() && !bytes.Equal(hash[:], report.MerkleRoot[:]) {
    return error
}
```

and set testMode to true only in the test. That way the package does also not need to export `SentinelRoot`

Reply:
Removed the sentinel value. Removed some randomness from the test setup so that the root can be calculated directly.